#2384 added test script in meta data source on new UI

Patrykb0802 opened the pull request:
Added validateScript() method

Added 'Run script' button to Meta Data Point editing

Added alert that shows result of working script

Modified script textarea to run validateScript method

Added script.runScript translation key to en.json

Added correct error message when creating new data point and script field is empty

Added delayed validation, so requests for validation are not send every letter typed

Added handling of error when field for script is empty and you press "run script", now when field is empty and you press "run script" nothing happens

Deleted duplicated property key dox.newUISettings

Patrykb0802 committed:
#2384 Added test script in Meta Data Source on new UI

In point.vue:
- Added validateScript() method
- Added 'Run script' button to  Meta Data Point editing
- Added alert that shows result of working script
- Modified script textarea to run validateScript method
- Added new rule that is checking if script is correct

- Added script.runScript translation key to en.json

Patrykb0802 committed:
#2384 Added test script in Meta Data Source on new UI

- Added correct error message when creating new data point and script field is empty

- Added delayed validation, so requests for validation are not send every letter typed

- Added handling of error when field for script is empty and you press "run script", now when field is empty and you press "run script" nothing happens

- Deleted duplicated property key dox.newUISettings

Patrykb0802 committed:
#2384 Added test script in Meta Data Source on new UI

Patrykb0802 committed:
Revert "#2384 Added test script in Meta Data Source on new UI"

This reverts commit 032e8e15dfdad7ad099d2c24afad737937466c32.

Patrykb0802 committed:
Revert "#2384 Added test script in Meta Data Source on new UI"

This reverts commit 23d8dbfab2fc619d7b4c176193b7bf484b9c2e5f.

Patrykb0802 committed:
#2384 added test script in meta data source on new UI

- Added validation in validateScript() method for empty text area of script

- Added script validation when focusing out of the script text area

- Added script validation when changing data point type so there is no possibility of saving with incorrect script

Limraj committed:
Merge branch 'release/2.7.6' into feature/#2384_Added_test_script_in_Meta_Data_Source_on_new_UI

Limraj committed:
Update en.json

corrected en.json

Limraj reviewed (commented):
Problem 1

1. Click on 'Create new data point' for Meta Data Source;
2. Created Data Point with script "return 1;" as dp_1; 
3. Refresh page and edit dp_1;
4. Added disabled Data Point to Script Context -> This does not trigger validation;
Expected behavior:
Also, a change in the context of the script should validate the script.

Problem 2

1. Click on 'Create new data point' for Meta Data Source;
2. Created Data Point with script "return 1;" as dp_1;
5. Refresh page, and edit Data Point dp_1;
6. Click on 'Run script' and add disabled Data Point to context, and click update -> Successfully updated script failing validation, as you can see by clicking in the Script field and then outside this field;

Problem 3

1. Click on 'Create new data point' for Meta Data Source;
2. Created Data Point with script "return 1;" as dp_1;
3. Refresh page, and edit Data Point dp_1;
4. Click on 'Run script' and add disabled Data Point to context, and click update;
7. Refresh page and edit dp_1;
8. Click on 'Run script' -> no message with the result of the script

Problem 4

1. Click on 'Create new data point' for Meta Data Source;
2. Created Data Point with script "return 1;" as dp_1;
3. Refresh page, and edit Data Point dp_1 -> The update button is not active, we change the description and it is still greyed out;

Expected behavior:
After changing fields such as description, or other field, the update button should be active;

Commit:
#2384 added test script in meta data source on new UI

- Added validation after adding context datapoint for script

- Added validation after editing text fields like data point name, datapoint export ID and description

Limraj reviewed (commented):
Problem 1

1. Click on 'Create new data point' for Meta Data Source;
2. Created Data Point with script "return 1;" as dp_1;
3. Refresh page and edit dp_1;
4. Added disabled Data Point to Script Context -> This does not trigger validation;

Expected behavior:
Also, a change in the context of the script should validate the script.

Problem 5

1. Click on 'Create new data point' for Meta Data Source;
2. Created Data Point with script "return 1;" as dp_1;
3. Refresh page, and edit Data Point dp_1 -> The update button is not active, is greyed -> error

Expected behavior:
As in other forms, the update button should be available after opening the point edit form

Problem 6
1. Click on 'Create new data point' for Meta Data Source;
2. Created Data Point with script "return 1;" as dp_1;
3. Click on 'Cancel' -> The form window does not close and validates the script;

Expected behavior:
Clicking the cancel button should reset the form values ​​and close the form window.

Patrykb0802 committed:
#2384 added test script in meta data source on new UI

- Moved error messages to alert component

- Added valdiation on removing context datapoint

Patrykb0802 committed:
#2384 added test script in meta data source on new UI

- Removed validation on focus out for textareas: Data point name, export ID, description

Limraj reviewed (commented):
Problem 2-4 -> ok;
Problem 1 -> Changing script context, triggers validation -> ok;
Problem 5 -> Right after opening the form edit, it unnecessarily performs validation -> to correct;
Problem 6 -> Clicking cancel does not trigger the validation again and closes the form window -> ok;
